pumpkin-protocol/Cargo.toml

```diff
@@ -14,7 +14,7 @@ serde.workspace = true
 thiserror.workspace = true
 itertools.workspace = true
 log.workspace = true
-
+tokio.workspace = true
```

Comment:
For what you need tokio here ?

Reply:
For the convenience functions provided by `AsyncReadExt` and `AsyncWriteExt` in tokio. This includes functions like read_u8, read_i32, write_i32, write_u16_le etc. Although thinking about it, adding tokio for convenience seems a little excessive. I'll probably remove tokio 👍

Reply:
On second thought, I'll probably keep tokio since it would overcomplicate the code. But if you feel like tokio shouldn't be in pumpkin-protocol, then I'll go ahead and get rid of it 👍